### ISSUE TITLE
Shall I move from Salvattore?

### ISSUE THREAD

**Issue by bserem** [attila/savvior#20]
I didn't knew about Savvior. Just found out about it.

Earlier this year I created a Drupal module for Salvattore and been running it ever since. Had some issues, one of them still remains (balanced height in columns) but I'm happy with it.

I'm thinking, shall I do the switch to Savvior? It should be straightforward to create a module for Savvior in Drupal. In terms of frontend performance is it any better than the competitors?

Sorry for the dump questions and thanks for the project ;)

**Comment by attila** [comment]
Savvior was initially developed to overcome some limitations of Masonry or Salvattore. It remained a relatively small library with a limited feature set although I still have plans to extend it in the future.

The algorithm for recalculating items in columns is very simple which gives varied results with items having variable heights so I am not sure if Savvior can offer an improvement here.

To be honest, I was thinking about developing a support module for Drupal myself.

I do not think there is a performance improvement over either Salvattore or Masonry although I have not conducted any tests yet. My advice is that you give it a go and play with it and whether you find any bugs or you have anything to feed back, let me know.

**Comment by bserem** [comment]
Let me try porting this for Drupal, I might ask you some questions though :)

I'll try to do this in the upcoming weekend.

**Comment by attila** [comment]
Sure, knock yourself out. Do you plan both a 7.x and a 8.x version with views integration? I can help with the implementation, author of [css_delivery](https://www.drupal.org/project/css_delivery) here.

**Comment by bserem** [comment]
First draft: https://www.drupal.org/project/savvior
that actually works in Drupal7. I'll build upon this to to see how it goes with some heavy pages my client uses.
You have git rights to the project.
It requires libraries, so no "savvior" code should exist there if I've done my homework right :)

I have tried to port Salvattore to D8, but I still can't get views to know about it (libraries etc are prepared). Once I manage to do that, Savvior will be a matter of minutes.

ps: I've got to try css_delivery!! Looks pretty promising.
